ui/src/main/java/org/jtaccuino/core/ui/api/CellData.java

```diff
@@ -155,8 +137,37 @@ public static OutputType of(String outputType) {
             }
         }
 
-        public static OutputData of(OutputType outputType, Map<String, String> mimeBundle) {
-            return new OutputData(outputType, mimeBundle);
+        public OutputType type();
+    }
+
+    public static record MimeTypeBasedOutputData(OutputType type, Map<String, String> mimeBundle) implements OutputData {
+
+        public static enum MimeType {
+            TEXT_PLAIN("text/plain"),
+            IMAGE_PNG("image/png");
+
+            private final String mimeType;
+
+            public String toMimeType() {
```

Comment:
Also naming is a bit confusing. You're working on the MimeType enumeration and ask it to turn into a MimeType. Huh?
`MimeType.IMAGE_PNG.toMimeType()`

Reply:
Correct. Trying to keep the change small. Something for another PR.

Reply:
This was just moved around.